connector/src/main/scala/com/microsoft/kusto/spark/utils/ContainersCache.scala

```diff
@@ -0,0 +1,42 @@
+package com.microsoft.kusto.spark.utils
+
+import com.microsoft.azure.kusto.data.Client
+import com.microsoft.kusto.spark.utils.CslCommandsGenerator.generateCreateTmpStorageCommand
+import com.microsoft.kusto.spark.utils.{KustoDataSourceUtils => KDSU}
+
+import org.joda.time.{DateTime, DateTimeZone, Period}
+
+import scala.collection.JavaConverters._
+
+class ContainersCache[A](val dmClient: Client, val clusterAlias: String, val command: String, cacheEntryCreator: ContainerAndSas => A) {
```

Comment:
`ContainerProvider` ?

Reply:
its a cache ... no ?

Reply:
Usually, a cache doesn't use round-robin to provide you with values... There is some business logic here. so cache seems like it's not providing enough information